source/Calamari/Kubernetes/Commands/KubernetesDeploymentCommandBase.cs

```diff
@@ -0,0 +1,131 @@
+#if !NET40
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Calamari.Aws.Integration;
+using Calamari.Commands;
+using Calamari.Commands.Support;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Behaviours;
+using Calamari.Common.Features.ConfigurationTransforms;
+using Calamari.Common.Features.ConfigurationVariables;
+using Calamari.Common.Features.Packages;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Features.StructuredVariables;
+using Calamari.Common.Features.Substitutions;
+using Calamari.Common.Plumbing.Deployment;
+using Calamari.Common.Plumbing.Deployment.Journal;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment;
+using Calamari.Deployment.Conventions;
+using Calamari.Kubernetes.Conventions;
+using Calamari.Kubernetes.Integration;
+
+namespace Calamari.Kubernetes.Commands
+{
+    public abstract class KubernetesDeploymentCommandBase  : Command
+    {
+        private readonly ILog log;
+        private readonly IDeploymentJournalWriter deploymentJournalWriter;
+        private readonly IVariables variables;
+        private readonly ICalamariFileSystem fileSystem;
+        private readonly IExtractPackage extractPackage;
+        private readonly ISubstituteInFiles substituteInFiles;
+        private readonly IStructuredConfigVariablesService structuredConfigVariablesService;
+        private readonly Kubectl kubectl;
+        private PathToPackage pathToPackage;
+
+        protected KubernetesDeploymentCommandBase(
+            ILog log,
+            IDeploymentJournalWriter deploymentJournalWriter,
+            IVariables variables,
+            ICalamariFileSystem fileSystem,
+            IExtractPackage extractPackage,
+            ISubstituteInFiles substituteInFiles,
+            IStructuredConfigVariablesService structuredConfigVariablesService,
+            Kubectl kubectl)
+        {
+            this.log = log;
+            this.deploymentJournalWriter = deploymentJournalWriter;
+            this.variables = variables;
+            this.fileSystem = fileSystem;
+            this.extractPackage = extractPackage;
+            this.substituteInFiles = substituteInFiles;
+            this.structuredConfigVariablesService = structuredConfigVariablesService;
+            this.kubectl = kubectl;
+
+            Options.Add("package=", "Path to the NuGet package to install.",
+                v => pathToPackage = new PathToPackage(Path.GetFullPath(v)));
+        }
+
+        protected abstract IEnumerable<IInstallConvention> CommandSpecificConventions();
+
+        public override int Execute(string[] commandLineArguments)
```

Comment:
most of the content of this method is exactly what is removed from `KubernetesApplyRawYamlCommand`.